src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http3RequestStream.cs

```diff
@@ -0,0 +1,1277 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Net.Http.Headers;
+using System.Net.Quic;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Runtime.CompilerServices;
+using System.Net.Http.QPack;
+using System.Runtime.ExceptionServices;
+using System.Buffers;
+
+namespace System.Net.Http
+{
+    internal sealed class Http3RequestStream : IHttpHeadersHandler, IAsyncDisposable, IDisposable
+    {
+        /// <summary>
+        /// Unknown frame types with a payload larger than this will result in tearing down the connection.
+        /// Frames smaller than this will be ignored and drained.
+        /// </summary>
+        private const int MaximumUnknownFramePayloadLength = 4096;
+
+        private readonly HttpRequestMessage _request;
+        private Http3Connection _connection;
+        private readonly long _streamId;
+        private QuicStream _stream;
+        private ArrayBuffer _sendBuffer;
+        private readonly ReadOnlyMemory<byte>[] _gatheredSendBuffer = new ReadOnlyMemory<byte>[2];
+        private ArrayBuffer _recvBuffer;
+        private TaskCompletionSource<bool> _expect100ContinueCompletionSource; // True indicates we should send content (e.g. received 100 Continue).
+
+        private CancellationTokenSource _goawayCancellationSource;
+        private CancellationToken _goawayCancellationToken;
+
+        // Allocated when we receive a :status header.
+        private HttpResponseMessage _response;
+
+        // Header decoding.
+        private QPackDecoder _headerDecoder;
```

Comment:
We should probably sync about how the QPackDecoder references the DynamicTable. From what I can tell, the Dynamic table will be per connection, so it may be tricky to get that all correct.

Reply:
I'm leaning towards removing `DynamicTable` from H3 completely. It's complex and not MVP -- we can add later once everything else is working.

Reply:
If you can confirm we can remove it entirely on both the client and server, then let's do it.